### PR TITLE
Add ability to search for courses starting soon.

### PR DESCRIPTION
**Pull request by mjfrey**
Add a new intent that searches for courses starting soon.

### CODE REVIEW

**Comment by mjfrey** [comment]
@clintonb please take a look.

File: enrollments/enrollments.py [hunk 1]
```diff
@@ -270,6 +282,38 @@ def about():
     return question(speech_text).reprompt(reprompt).simple_card(APP_NAME, speech_text)
 
 
+@ask.intent('EdXStartSoonIntent')
+def searchStartSoon():
+    courses = search_catalog_starting_soon()
+    count = len(courses)
+
+    if not count:
+        speech_text = 'Did not find any courses that are starting soon.'
```
Comment (clintonb) [inline enrollments/enrollments.py]: There are no courses starting within the next two weeks.

File: enrollments/enrollments.py [hunk 2]
```diff
@@ -270,6 +282,38 @@ def about():
...
+        word = 'Here they are'
+    else:
+        word = 'Here are the first two'
+
+        speech_text = '<speak><p>I found {count} courses starting soon</p> {word}'.format(count=count, word=word)
+
+        for course in courses[:2]:
+            speech_text += '<p>{}</p>'.format(course['title'])
+
+        if len(courses) <= 2:
```
Comment (clintonb) [inline enrollments/enrollments.py]: De-indent this block.

File: enrollments/enrollments.py [hunk 3]
```diff
@@ -270,6 +282,38 @@ def about():
...
+    if count == 1:
+        word = 'The course starting soon is'
+    elif count == 2:
+        word = 'Here they are'
+    else:
+        word = 'Here are the first two'
+
+        speech_text = '<speak><p>I found {count} courses starting soon</p> {word}'.format(count=count, word=word)
+
+        for course in courses[:2]:
```
Comment (clintonb) [inline enrollments/enrollments.py]: De-indent this block.

File: enrollments/enrollments.py [hunk 4]
```diff
@@ -270,6 +282,38 @@ def about():
...
+def searchStartSoon():
+    courses = search_catalog_starting_soon()
+    count = len(courses)
+
+    if not count:
+        speech_text = 'Did not find any courses that are starting soon.'
+        return statement(speech_text)
+
+    if count == 1:
+        word = 'The course starting soon is'
```
Comment (clintonb) [inline enrollments/enrollments.py]: `'I found one course. It is '`

File: enrollments/enrollments.py [hunk 5]
```diff
@@ -270,6 +282,38 @@ def about():
...
+    elif count == 2:
+        word = 'Here they are'
```
Comment (clintonb) [inline enrollments/enrollments.py]: `'I found two courses. They are '`

File: enrollments/enrollments.py [hunk 6]
```diff
@@ -270,6 +282,38 @@ def about():
...
+    else:
+        word = 'Here are the first two'
```
Comment (clintonb) [inline enrollments/enrollments.py]: `'I found {count} courses starting soon. The first two are '`

File: enrollments/speech_assets/SampleUtterances.txt [hunk 7]
```diff
@@ -25,3 +25,7 @@ EdXSearchIntent courses on {Subject}
 EdXSearchIntent courses about {Subject}
 EdXSearchIntent search for courses about {Subject}
 EdXSearchIntent search for courses on {Subject}
+EdXStartSoonIntent find courses that are starting soon
+EdXStartSoonIntent find courses that are starting shortly
+EdXStartSoonIntent search for courses that are starting soon
+EdXStartSoonIntent search for courses that are starting shortly
```
Comment (clintonb) [inline enrollments/speech_assets/SampleUtterances.txt]: Other options:
what courses are starting soon
which courses are starting soon
what courses are starting shortly
which courses are starting shortly

**Pull request merged**